### PR TITLE
Add fcntl OFD commands for macOS

### PR DESCRIPTION
**Pull request by anacrolix**
I also followed all the PR instructions and sorted the semver for apple.

### CODE REVIEW

**Comment by anacrolix** [comment]
So I noticed that the tests run on macos-13, and macos-14 is available (possibly needed for the OFD feature, despite the constants being 4 years old according to the kernel changelog), and also that the tests only run on Intel chips. I do not think it would be too much work to add aarch64 to the macos testing (assuming GitHub has a runner for this).

**Comment by JohnTitor** [comment]
Could you rebase onto the latest main and squash commits into one?

**Comment by anacrolix** [comment]
Yes, I'll do that soon, thanks.

**Comment by anacrolix** [comment]
Okay, I've done that. It's extraordinarily painful, any reason you can't squash on merge?

**Comment by anacrolix** [comment]
I don't think my change caused the macOS CI failure, I'm not sure what to do about that one.

**Comment by JohnTitor** [comment]
> I don't think my change caused the macOS CI failure, I'm not sure what to do about that one.

You updated the macOS version and it introduced some breaking changes, you have to account for that.



> Okay, I've done that. It's extraordinarily painful, any reason you can't squash on merge?

We use the merge queue and it doesn't allow me to select the merge method sadly.

**Comment by anacrolix** [comment]
It'll take someone more knowledgable about macos to fix this I think. I didn't find anything straightforward to resolve it.

**Comment by anacrolix** [comment]
I found https://github.com/rust-lang/libc/pull/3575 but it's mysteriously closed.

**Comment by anacrolix** [comment]
This includes https://github.com/rust-lang/libc/pull/3578 and should be merged after that.

**Comment by tgross35** [comment]
@anacrolix please rebase and drop the commits that are no longer needed (1a8e6f490df340ab080ca3507fc91178de7dfc5a, cb7d5b13877695b6ee082a59ea712fa431005cb7).

@rustbot author

**Comment by anacrolix** [comment]
Roger

**Comment by anacrolix** [comment]
@tgross35 Done

File: .github/workflows/full_ci.yml [hunk 1]
```diff
@@ -1,9 +1,6 @@
 name: full CI
 
-on:
-  merge_group:
-  pull_request:
-    types: [opened, synchronize, reopened]
+on: [push]
```
Comment (tgross35) [inline .github/workflows/full_ci.yml:3]: What is the intent with this change? This commit should be dropped if it is only for testing.

Comment (anacrolix) [reply .github/workflows/full_ci.yml]: I removed this. It was frustrating having to make a PR to get CI to run. Is that a worthwhile change in another PR?

**Comment by tgross35** [comment]
There is also an empty change to `ci/style.sh`, can you drop that?

File: .github/workflows/full_ci.yml (hunk 1 above)
Comment (tgross35) [reply .github/workflows/full_ci.yml]: Making a PR to get CI to run, as opposed to what exactly? Usually CI is only run on PRs and after merges.

**Review by tgross35 — APPROVED** [review]
Thanks!

**Pull request merged**